src/QsFmt/SPECIFICATION.md

```diff
@@ -0,0 +1,68 @@
+﻿# Q# Formatter Design Specification Document
+
+QsFmt is a source code formatter for Q#.
+
+## Where to get the formatter
+
+The `Microsoft.Quantum.QSharp.Formatter` NuGet package will be distributed with the rest of Q# on NuGet.org.
+As a dotnet tool, it is installed with the command `dotnet tool install Microsoft.Quantum.QSharp.Formatter`.
+
+## How to use the command line formatter
+
+After installing the tool the command `qsfmt` is used to run the formatter.
+There are two commands supported for the tool: 'format' and 'update'.
+The 'format' command runs the tool as a formatter for Q# code, running only those transformation rules that update whitespace and affect indentation. The underlying code will not be changed.
+The 'update' command allows the tool to be used as a means of updating old Q# code, replacing deprecated syntax with newer supported syntax. These transformation rules change the actual code given to them, and may occasionally fail.
+These two commands work separately and can't be run together. If the user wishes to both format and update their code, they would need to run the tool twice, first with 'update' and then with 'format'.
+Both of these commands should be idempotent, meaning that iterated uses of the commands will not continue to have effects on the inputs.
+
+The tool will initially only support the 'update' command, as the formatting functionality is still being worked on. The 'format' command will be unavailable until it is supported.
+
+## Input and Output
+Input files to the formatter will be the last argument(s) to the tool.
+At least one input argument is expected. If no inputs are given, usage information is printed.
+Inputs can be given as one to many paths to directories, files, or any combination. In directories specified, all files with the '.qs' extension found will be taken as input.
+Basic wild cards may be used in the paths.
+The `--recurse` option can be specified to process the input directories recursively for input files. You can specify this for all input directories, not per-directory.
```

Comment:
It looks like some of these have different names than the corresponding `dotnet-format` options documented at https://github.com/dotnet/format, would it perhaps make sense to align there? Also, should there be a `--check` mode that can be used in CI to enforce formatting?

Reply:
I was using Fantomas as an example for this, found here : https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md.
Could you be more specific with what changes you are looking for?
The `--check` option is a good possibility.

Reply:
I recommend modeling after https://github.com/dotnet/format because other Q# commands such as  build and run use .NET.

Reply:
Moving this conversation to the spec document comments section. Please pick up this conversation there.